Marlin/ultralcd.cpp

```diff
@@ -684,11 +684,13 @@ void kill_screen(const char* lcd_msg) {
 
   #if ENABLED(MENU_ITEM_CASE_LIGHT)
 
-    extern bool case_light_on;
+    extern uint8_t case_light_brightness;  
+    bool case_light_on = (case_light_brightness >= 128) ? true : false;
     extern void update_case_light();
 
     void toggle_case_light() {
       case_light_on = !case_light_on;
```

Comment:
You can drop this line if you have `update_case_light` handle updating the state of `case_light_on` based on `case_light_brightness`.